### PR TITLE
Remove the "tmms" entry in the txpk object when the entry is null

### PR DESCRIPTION
**Pull request by mullerch**
The [Semtech UDP protocol states](https://github.com/Lora-net/packet_forwarder/blob/master/PROTOCOL.TXT#L134) : 

> The root object can contain an array named "rxpk":
> 
> ``` json
> {
> 	"rxpk":[ {...}, ...]
> }
> ```
> 
> That array contains at least one JSON object, each object contain a RF packet 
> and associated metadata with the following fields:
> 
>  Name |  Type  | Function
> :----:|:------:|--------------------------------------------------------------
>  time | string | UTC time of pkt RX, us precision, ISO 8601 'compact' format
>  tmms | number | GPS time of pkt RX, number of milliseconds since 06.Jan.1980
>  tmst | number | Internal timestamp of "RX finished" event (32b unsigned)
>  freq | number | RX central frequency in MHz (unsigned float, Hz precision)
>  chan | number | Concentrator "IF" channel used for RX (unsigned integer)
>  rfch | number | Concentrator "RF chain" used for RX (unsigned integer)
>  stat | number | CRC status: 1 = OK, -1 = fail, 0 = no CRC
>  modu | string | Modulation identifier "LORA" or "FSK"
>  datr | string | LoRa datarate identifier (eg. SF12BW500)
>  datr | number | FSK datarate (unsigned, in bits per second)
>  codr | string | LoRa ECC coding rate identifier
>  rssi | number | RSSI in dBm (signed integer, 1 dB precision)
>  lsnr | number | Lora SNR ratio in dB (signed float, 0.1 dB precision)
>  size | number | RF packet payload size in bytes (unsigned integer)
>  data | string | Base64 encoded RF packet payload, padded

- So tmms should be mandatory. 
- However, the examples shown are all missing the tmms field.
- There is no example of a null field, but it should be valid in JSON.

So most correct would certainly be setting to null, as it's done now. But as it's not clear, to maximize interoperability, I suggest not setting the tmms field if value is null. In v1.3 of the protocol, this field did not exist. So retro-compatibility of implementations will guarantee it works.

What do you think is best?

### CODE REVIEW

**Comment by brocaar** [comment]
Hi @mullerch the tests are failing because you added the line to `tmst`, which is not an `Option` type.
In general I'm happy to leave `null` fields out the encoded output.

File: src/structs.rs [hunk 1]
```diff
@@ -187,6 +187,7 @@ pub struct RXPK {
     /// GPS time of pkt RX, number of milliseconds since 06.Jan.1980
     pub tmms: Option<u64>,
```
Comment (brocaar) [inline src/structs.rs:190]: I believe you intended to add it above this field.

**Comment by brocaar** [comment]
Could you do the same for the `codr` and `lsnr` fields?

**Comment by mullerch** [comment]
TXPK also has Option fields but it's never serialized for the LNS so I guess it's not an issue. If you can confirm Option it will handle both `null` and non-existent entries for de-serialization.

**Comment by brocaar** [comment]
> If you can confirm Option it will handle both null and non-existent entries for de-serialization.

I believe this is indeed the case, as some of the fields are omitted by the ChirpStack Gateway Bridge (e.g. `prea` is not set by default).

**Pull request merged**